src/codeu/chat/server/Model.java

```diff
@@ -68,12 +73,27 @@ public int compare(Time a, Time b) {
   private final Uuid.Generator userGenerations = new LinearUuidGenerator(null, 1, Integer.MAX_VALUE);
   private Uuid currentUserGeneration = userGenerations.make();
 
+  private final String USERS = "users";
+  private final String MESSAGES = "messages";
+  private final String CONVERSATIONS = "conversations";
+
+  // FireBase database reference for class
+  private static DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+
   public void add(User user) {
     currentUserGeneration = userGenerations.make();
 
     userById.insert(user.id, user);
     userByTime.insert(user.creation, user);
     userByText.insert(user.name, user);
+
+    DatabaseReference usersRef = database.child(USERS);
+
+    Map<String, Object> userMap  = new HashMap<>();
+    userMap.put(user.id.toString() + "/" + "name", user.name.toString());
```

Comment:
Makes sense but is there a particular reason not to set it up such that the key is only the uuid, and the value is another map mapping property name to value? So for example, 
{uuid: 
  'name': ...
  'creation': ...
}